### PR TITLE
Feature/generate topology

### PR DESCRIPTION
**Pull request by sungminlee114**
I've modified some features of `generate_topology.py`. Specifically,
1) Added fixed seed policy
2) Added changing tx and rx ratio of a simulation

Consequently, I've changed the readme.md file according to the changes in the `generate_topology.py`.

Also, I found that some of the path settings in the readme file is not consistent. So I've changed those too.

### CODE REVIEW

File: scripts/generate-topology.py [hunk 1]
```diff
@@ -19,17 +20,42 @@ def main():
     p.add_argument('-i', dest='input', required=True)
     p.add_argument('-o', dest='output', required=True)
     p.add_argument('-c', dest='count', type=int, default=1)
-    p.add_argument('--generated-seed', dest='generated_seed', action='store_true', default=False)
+    p.add_argument('--seed', dest='seed_policy', type=str, default="r", \
+        help="Set seed policy. 'r' for random seed, 'g' for generated seed, 'f' for fixed seed")
```
Comment (nfi) [inline scripts/generate-topology.py]: ArgParser supports choices, making any further error checking unnecessary. 

`p.add_argument('--seed', dest='seed_policy', choices=['r', 'g', 'f'], default='r',`

The backslash is not needed, and the next line should be indented further (build warning: "PEP 8: E128 continuation line under-indented for visual indent").

File: scripts/generate-topology.py [hunk 2]
```diff
@@ -19,17 +20,42 @@ def main():
...
     p.add_argument('--topology', dest='topology', default=None)
     p.add_argument('--min-distance', dest='min_distance', type=int, default=0)
+    p.add_argument('--tx_rat', dest='tx_ratio', type=float, default=1.0, nargs='+', help="Transmision ratio of network. Values will be rounded up to 2 digits.")
+    p.add_argument('--rx_rat', dest='rx_ratio', type=float, default=1.0, nargs='+', help="Receive ratio of network. Values will be rounded up to 2 digits.")
```
Comment (nfi) [inline scripts/generate-topology.py]: - The existing arguments use dashes and not underscores. What do you think about naming the new arguments `--tx-ratio` and `--rx-ratio` ?
- Typo: "Transmision" => "Transmission".
- Long lines (build warning: "PEP 8: E501 line too long (160 > 120 characters)")

File: scripts/generate-topology.py [hunk 3]
```diff
@@ -19,17 +20,42 @@ def main():
...
+    
+    args.seed_policy = args.seed_policy.lower()
+    if args.seed_policy not in ['r', 'g', 'f']:
+        print(f"Seed policy must be one of 'r', 'g', 'f' ({args.seed_policy} was given)")
+        sys.exit("Invalid seed policy")
+    
+    if type(args.tx_ratio) == float:
+        args.tx_ratio = [args.tx_ratio]
+    if type(args.rx_ratio) == float:
+        args.rx_ratio = [args.rx_ratio]
```
Comment (nfi) [inline scripts/generate-topology.py]: These arguments are already lists as they are defined with `nargs='+'`. No conversion should be needed.

File: scripts/generate-topology.py [hunk 4]
```diff
@@ -47,21 +73,16 @@ def main():
         promote_multihop = False
 
     output_file = args.output
```
Comment (nfi) [inline scripts/generate-topology.py]: This variable does not seem to be used anymore and could be removed.

File: scripts/generate-topology.py [hunk 5]
```diff
@@ -47,21 +73,16 @@ def main():
...
-
+
```
Comment (nfi) [inline scripts/generate-topology.py]: Trailing whitespace.

File: scripts/generate-topology.py [hunk 6]
```diff
@@ -47,21 +73,16 @@ def main():
...
 
-        if args.count > 1:
-            output_file = f'{os.path.splitext(args.output)[0]}-{i + 1:05}.csc'
-        if os.path.exists(output_file):
-            print(f'Warning: the output file {output_file} already exists - skipping generation')
-            continue
-        print(f'Generating {output_file}')
-
-        if not args.generated_seed:
+        if  args.seed_policy == 'r':
```
Comment (nfi) [inline scripts/generate-topology.py]: Double spaces after `if`.

**Review by nfi — COMMENTED** [review]
Looks good. I only have some minor comments, mainly regarding code style.

**Comment by sungminlee114** [comment]
Great. Overall, I wasn't much concerned about the PEP but I'll get my PEP linters prepared for next prs. Thank you very much!

**Pull request merged**